docs/docs/addons-test-utils.md

```diff
@@ -0,0 +1,325 @@
+---
+id: test-utils
+title: Test Utilities
+permalink: docs/test-utils.html
+layout: docs
+category: Reference
+prev: shallow-compare.html
+next: perf.html
+---
+
+**Importing**
+
+```javascript
+import ReactTestUtils from 'react-addons-test-utils' // ES6
+var ReactTestUtils = require('react-addons-test-utils') // ES5 with npm
+var ReactTestUtils = React.addons.TestUtils; // ES5 with react-with-addons.js
+```
+
+## Overview
+
+`ReactTestUtils` makes it easy to test React components in the testing framework of your choice. At Facebook we use [Jest](https://facebook.github.io/jest/) for painless JavaScript testing. Learn how to get started with Jest through the Jest website's [React Tutorial](http://facebook.github.io/jest/docs/tutorial-react.html#content).
+
+> Note:
+>
+> Airbnb has released a testing utility called Enzyme, which makes it easy to assert, manipulate, and traverse your React Components' output. If you're deciding on a unit testing utility to use together with Jest, or any other test runner, it's worth checking out: [http://airbnb.io/enzyme/](http://airbnb.io/enzyme/)
+
+ - [`Simulate`](#simulate)
+ - [`renderIntoDocument()`](#renderintodocument)
+ - [`mockComponent()`](#mockcomponent)
+ - [`isElement()`](#iselement)
+ - [`isElementOfType()`](#iselementoftype)
+ - [`isDOMComponent()`](#isdomcomponent)
+ - [`isCompositeComponent()`](#iscompositecomponent)
+ - [`isCompositeComponentWithType()`](#iscompositecomponentwithtype)
+ - [`findAllInRenderedTree()`](#findallinrenderedtree)
+ - [`scryRenderedDOMComponentsWithClass()`](#scryrendereddomcomponentswithclass)
+ - [`findRenderedDOMComponentWithClass()`](#findrendereddomcomponentwithclass)
+ - [`scryRenderedDOMComponentsWithTag()`](#scryrendereddomcomponentswithtag)
+ - [`findRenderedDOMComponentWithTag()`](#findrendereddomcomponentwithtag)
+ - [`scryRenderedComponentsWithType()`](#scryrenderedcomponentswithtype)
+ - [`findRenderedComponentWithType()`](#findrenderedcomponentwithtype)
+
+### Shallow Rendering
+
+Shallow rendering lets you render a component "one level deep" and assert facts about what its render method returns, without worrying about the behavior of child components, which are not instantiated or rendered. This does not require a DOM.
+
+ - [`createRenderer()`](#createrenderer)
+ - [`shallowRenderer.render()`](#shallowrenderer.render)
+ - [`shallowRenderer.getRenderOutput()`](#shallowrenderer.getrenderoutput)
+
+Call [`createRenderer()`](#createrenderer) in your tests to create a shallow renderer. You can think of this as a "place" to render the component you're testing, where it can respond to events and update itself.
```

Comment:
>You can think of this as a "place" to render the component you're testing, where it can respond to events and update itself.

AFAIK it can't. Shallow rendering doesn't use event system.

I would just leave

>You can think of this as a "place" to render the component you're testing, and from which you can extract the component's output.